coreos_prometheus/Tiltfile

```diff
@@ -0,0 +1,183 @@
+# (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+
+def _find_root_tiltfile_dir():
+    # Find top-level Tilt path
+    current = os.path.abspath('./')
+    while current != '/':
+        if os.path.exists(os.path.join(current, '')):
+            return current
+
+        current = os.path.dirname(current)
+
+    fail('Could not find root Tiltfile')
+
+def _find_cache_dir():
+    from_env = os.getenv('TILT_CACHE_DIR', '')
+
+    if from_env != '':
+        return from_env
+
+    cachedir = os.path.join(_find_root_tiltfile_dir(), '.tilt-cache')
+    os.putenv('TILT_CACHE_DIR', cachedir)
+
+    return cachedir
+
+def name(c):
+    return c['metadata']['name']
+
+def replace_target(src, target):
+    if not os.path.exists(src):
+        fail("src path '%s' not found!" % src)
+
+    if not os.path.exists(target):
+        fail("target path '%s' not found!" % target)
+
+    local("cp %s %s" % (src, target))
+
+def download_files():
+    cachedir = _find_cache_dir()
+    if not os.path.exists(cachedir):
+        local('mkdir %s' % cachedir, quiet=True)
+
+    kube_prometheus_version = 'v0.5.0'
+    kube_prometheus_tarball = os.path.join(cachedir, 'coreos-kube-prometheus-%s.tar.gz' % kube_prometheus_version)
+
+    extract_path = os.path.join(cachedir, 'prometheus-operator-kube-prometheus-%s' % kube_prometheus_version)
+
+    if os.path.exists(kube_prometheus_tarball) and os.path.exists(extract_path):
+        return extract_path
+
+    local("curl -L https://api.github.com/repos/coreos/kube-prometheus/tarball/%s -o %s.tmp" % (kube_prometheus_version, kube_prometheus_tarball))
+
+    local("mkdir %s" % extract_path)
+    local("tar -C %s -xzf %s.tmp --strip-components=1 --wildcards 'prometheus-operator-kube-prometheus-*/manifests'" % (extract_path, kube_prometheus_tarball))
+
+    # only copy file to final location to prevent redoing the above if the above
+    # have completed successfully.
+    local("mv %s.tmp %s" % (kube_prometheus_tarball, kube_prometheus_tarball))
+
+    # return path containing manifests directory
+    return extract_path
+
+def patch_prometheus(path):
+    manifestsPath = os.path.join(path, 'manifests')
+    prometheusPatchesPath = os.path.join(os.path.dirname(__file__), 'prometheus')
+
+    # patches to modify certain behaviour
+    #  modify clusterrole for greater access
+    #  ensure all namespaces are scanned for alerts
+    localFiles = listdir(prometheusPatchesPath)
```

Comment:
Could probably read in the yaml and modify the resulting objects instead of relying on files, depends on whether it's easier for someone to provide an additional path to check for patches, or if should expect objects defined in Tiltfile to be passed to manage this.